packages/webapp/src/containers/Finances/AddSale/index.jsx

```diff
@@ -1,110 +1,98 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
 import React from 'react';
-import CropSaleForm from '../../../components/Forms/CropSale';
-import GeneralRevenueForm from '../../../components/Forms/GeneralRevenue';
-import { addOrUpdateSale } from '../actions';
-import { userFarmSelector, measurementSelector } from '../../userFarmSlice';
-import { currentAndPlannedManagementPlansSelector } from '../../managementPlanSlice';
+import GeneralRevenue from '../../../components/Forms/GeneralRevenue';
+import useCropSaleInputs from '../useCropSaleInputs';
+import { addSale } from '../actions';
+import { userFarmSelector } from '../../userFarmSlice';
 import { useDispatch, useSelector } from 'react-redux';
 import { useTranslation } from 'react-i18next';
 import { useCurrencySymbol } from '../../hooks/useCurrencySymbol';
 import { hookFormPersistSelector } from '../../hooks/useHookFormPersist/hookFormPersistSlice';
 import { HookFormPersistProvider } from '../../hooks/useHookFormPersist/HookFormPersistProvider';
-import { revenueTypeSelector } from '../../revenueTypeSlice';
-import { getRevenueFormType } from '../util';
-import { revenueFormTypes as formTypes } from '../constants';
+import { revenueTypeByIdSelector, revenueTypeTileContentsSelector } from '../../revenueTypeSlice';
 
 function AddSale() {
-  const { t } = useTranslation();
+  const { t } = useTranslation(['translation', 'revenue']);
   const dispatch = useDispatch();
 
-  const managementPlans = useSelector(currentAndPlannedManagementPlansSelector) || [];
   const farm = useSelector(userFarmSelector);
-  const system = useSelector(measurementSelector);
   const persistedFormData = useSelector(hookFormPersistSelector);
   const { revenue_type_id } = persistedFormData || {};
-  const revenueType = useSelector(revenueTypeSelector(revenue_type_id));
+  const revenueType = useSelector(revenueTypeByIdSelector(revenue_type_id));
+  const revenueTypes = useSelector(revenueTypeTileContentsSelector);
+
+  const revenueTypeReactSelectOptions = revenueTypes?.map((type) => {
+    const retireSuffix = type.deleted ? ` ${t('REVENUE.EDIT_REVENUE.RETIRED')}` : '';
 
-  const formType = getRevenueFormType(revenueType);
+    return {
+      value: type.revenue_type_id,
+      label: type.farm_id
+        ? type.revenue_name + retireSuffix
+        : t(`revenue:${type.revenue_translation_key}`),
+    };
+  });
```

Comment:
This logic seems to be the same both for the "adding" and the "editing" versions -- maybe instead of making it a prop we could bake this in the form component itself?

Reply:
They are slightly different -- Edit sale adds in any "soft deleted" expense types -- but I like the idea. 

I am just trying to process two things 
- if we should be adding logic to the component or if it should stay in the container. 
- And also if it makes sense that options are configurable at the container level 

If it just that it is reused maybe I could just put the reused function in the util.js ?

Reply:
@Duncan-Brain ohhh, thanks for clarifying! I hadn't noticed that one. I don't think there's necessarily a hard fast rule why you shouldn't be adding logic to the component. If the component was generic and domain agnostic (part of the design system) then I'd say yeah, let's keep the logic outside of it so that it stays as generic and reusable as possible, but in this case the `GeneralRevenue` component is already tied to a certain domain and business logic, so it's not entirely abstract. That being said, either way works! A util would also work, the goal would be not to have to maintain this piece of logic in both places in the future.

Reply:
Okay I think I will add that utility in a follow up commit with any submit button decisions.